### PR TITLE
Add a build and run Dockerfile

### PR DESCRIPTION
**Pull request by rorpage**
This PR adds a Dockerfile that can build and run the CLI using a multi-stage build. I have tested this on Docker for OS X.

Thanks!

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# build
+FROM golang:stretch AS build-env
```
Comment (jmattheis) [inline Dockerfile]: Could you use a tag with a Go version?

Comment (rorpage) [reply Dockerfile]: Done!

**Comment by rorpage** [comment]
I was having trouble building and running the CLI locally, so I thought making a consistent build and run pipeline would be helpful.

When I built locally, I would get `ERROR user: Current not implemented on linux/amd64` when running the CLI and putting it in a container that could specify that for me helped.

**Comment by rorpage** [comment]
I'm happy to push it to the Docker hub as `rorpage/gotify-cli` if you like.

**Pull request merged**